pkg/services/ngalert/state/manager.go

```diff
@@ -170,19 +170,30 @@ func (st *Manager) ProcessEvalResults(ctx context.Context, alertRule *ngModels.A
 	return states
 }
 
-//nolint:unused
-func (st *Manager) newImage(ctx context.Context, alertRule *ngModels.AlertRule, state *State) error {
-	if state.Image == nil {
-		image, err := st.imageService.NewImage(ctx, alertRule)
-		if errors.Is(err, screenshot.ErrScreenshotsUnavailable) {
-			// It's not an error if screenshots are disabled.
-			return nil
-		} else if err != nil {
-			st.log.Error("failed to create image", "error", err)
-			return err
-		}
-		state.Image = image
+// Maybe take a screenshot. Do it if:
+// 1. The alert state is transitioning into the "Alerting" state from something else.
+// 2. The alert state has just transitioned to the resolved state.
+// 3. The state is alerting and there is no screenshot annotation on the alert state.
+func (st *Manager) maybeNewImage(ctx context.Context, alertRule *ngModels.AlertRule, state *State, oldState eval.State) error {
+	shouldScreenshot := state.Resolved ||
+		state.State == eval.Alerting && oldState != eval.Alerting ||
+		state.State == eval.Alerting && state.Image == nil
+	if !shouldScreenshot {
+		return nil
+	}
+
+	img, err := st.imageService.NewImage(ctx, alertRule)
+	if err != nil &&
+		errors.Is(err, screenshot.ErrScreenshotsUnavailable) ||
+		errors.Is(err, image.ErrNoDashboard) ||
+		errors.Is(err, image.ErrNoPanel) {
+		// It's not an error if screenshots are disabled, or our rule isn't allowed to generate screenshots.
+		return nil
```

Comment:
Do we log this condition? Especially the dashboard and panel one?

Reply:
I don't think so, and I don't think we should do it here. Any rules created in the query builder without connecting them to a panel will "fail" this check routinely and that shouldn't be logged because it's not an abnormal condition.